hack/community/generate.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# Given the current WMCO manifests, generate new community manifests.
+
+# Must be ran within WMCO root.
+
+# Example:
+# Cut community v5.1.0 for community-4.10
+# Run: bash ./hack/community/generate.sh 5.1.0 community-4.10
+
+# clone community-operators-prod repo
+clone_community() {
+  local CO_GIT="https://github.com/${CO_REPO}.git"
+
+  git clone "${CO_FORK}" "${CO_DIR}"
+  pushd "${CO_DIR}"
+  git remote add upstream "${CO_GIT}"
+  git fetch upstream main:upstream/main
+  git checkout upstream/main
+  git config user.name "${GITHUB_NAME}"
+  git config user.email "${GITHUB_EMAIL}"
+  echo "   ${CO_PROJECT} cloned"
+  popd
+}
+
+# replace necessary fields with the yq tool
+replace() {
+    export CREATED_AT=$(date +"%Y-%m-%dT%H:%M:%SZ")
+    export VALUE="community-windows-machine-config-operator"
+    export NAME="${VALUE}-${OPERATOR_VERSION}"
+    export CO_DESCRIPTION=${DESCRIPTION}
+    export DISPLAY_NAME="Community Windows Machine Config Operator"
+    export IMAGE=${QUAY_IMAGE}
+    export MATURITY="preview"
+    export VERSION="${OPERATOR_VERSION}"
+
+    yq eval --exit-status --inplace '
+      .metadata.annotations.createdAt |= strenv(CREATED_AT) |
+      .metadata.name |= strenv(NAME) |
+      .spec.description |= strenv(CO_DESCRIPTION) |
+      .spec.displayName |= strenv(DISPLAY_NAME) |
+      .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[2].value |= strenv(VALUE) |
+      .spec.install.spec.deployments[0].spec.template.spec.containers[0].image |= strenv(QUAY_IMAGE) |
+      .spec.maturity |= strenv(MATURITY) |
+      .spec.version |= strenv(VERSION)
+    ' "${CO_CSV}"
+}
+
+# copy the WMCO bundle and its contents to a new community folder. Rename files
+# as needed.
+generate_manifests() {
+  local CO_OPERATOR_DIR="${CO_DIR}/operators/community-windows-machine-config-operator"
+
+  echo "Update operator manifests"
+  mkdir "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}"
+  cp -r "${OPERATOR_MANIFESTS}" "${OPERATOR_METADATA}" "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}"
+  mv "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/windows-machine-config-operator.clusterserviceversion.yaml" "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/community-windows-machine-config-operator.${OPERATOR_VERSION}.clusterserviceversion.yaml"
+  pushd "${CO_DIR}"
+  git checkout -B "${OPERATOR_VERSION}"
+  CO_CSV="${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/community-windows-machine-config-operator.${OPERATOR_VERSION}.clusterserviceversion.yaml"
+
+  replace "$CO_CSV" "$OPERATOR_VERSION" "$DESCRIPTION" "$QUAY_IMAGE"
+}
+
+# create a signed commit for the community-operators-prod repo
+commit() {
+  TITLE="operators community-windows-machine-config-operator.v${OPERATOR_VERSION}"
+  MESSAGE="Signed of by: ${GITHUB_NAME} ${GITHUB_EMAIL}"
+
+  echo "commit changes"
+  git add --all
+  git commit -s -m "${TITLE}
+  ${MESSAGE}"
+  popd
```

Comment:
It is not clear what directory are you poping here.

Reply:
the tmp dir

Reply:
the push and pop needs to be contained within the same function, or done in the 'main' portion of the script. Written like this you've created a dependency that this is always called after the function that pushds.

and like @jrvaldes has said its not clear why this needs to be done without the pushd in the current function